### PR TITLE
Verify insert lines

### PR DESCRIPTION
**Pull request by MarkGibbons**
### Description

Adds the safe option to avoid repeatedly insert lines.

### Issues Resolved

Reported in issue #131. 

### Contribution Check List

- [ ] All tests pass.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

---
This change is [Reviewable](https://reviewable.io/reviews/sous-chefs/line/133)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,10 @@
 # line Cookbook CHANGELOG
 
-# Unreleased
+# Unreleased - for target version 2.3.0 in mid Janary 2018
```
Comment (JohnRoesler) [inline CHANGELOG.md]: Should this be January 2019?

Comment (MarkGibbons) [reply CHANGELOG.md]: yes

File: libraries/filter_helper.rb [hunk 2]
```diff
@@ -46,6 +46,35 @@ def lines_between(current, start, next_match)
...
+        @options[key.to_sym] = setting
+      end
+      @options
+    end
+
+    def safe_default
+      defined?(@safe) ? @safe : true
+    end
+
+    def safe_default=(option)
```
Comment (damacus) [inline libraries/filter_helper.rb:67]: What's the difference between this and a default value on the resource?

File: CHANGELOG.md (hunk 1 above)
Comment (MarkGibbons) [reply CHANGELOG.md]: Updated.

File: spec/unit/library/filter/after_spec.rb [hunk 3]
```diff
@@ -97,4 +98,16 @@
   it 'should insert a string' do
     expect(@filt.after(@solo_middle, [@pattern_c1, 'string1', :last])).to eq(%w(linef c1 string1 lineg))
   end
+
+  it 'should insert a line that matches the pattern by default, nil implies safe' do
```
Comment (JohnRoesler) [inline spec/unit/library/filter/after_spec.rb]: Should not?

File: libraries/filter_helper.rb (hunk 2 above)
Comment (MarkGibbons) [reply libraries/filter_helper.rb:67]: This code (safe_default and safe_default=) are the methods the filter resource uses to pass the setting for safe to the Filter class object.   I removed the code that sets safe to true and use the default or setting from the resource.

File: spec/unit/library/filter/after_spec.rb (hunk 3 above)
Comment (MarkGibbons) [reply spec/unit/library/filter/after_spec.rb]: Thanks, good catch.

File: libraries/filter_helper.rb (hunk 2 above)
Comment (MarkGibbons) [reply libraries/filter_helper.rb:67]: Line 63 isn't "defined?(@safe) ? @safe : true"  anymore, the only default for safe is set in the resource.  Thanks for the suggestion.

File: spec/unit/library/filter/after_spec.rb (hunk 3 above)
Comment (MarkGibbons) [reply spec/unit/library/filter/after_spec.rb]: Pushed a fix.

**Review by JohnRoesler — APPROVED** [review]
Lgtm

**Comment by MarkGibbons** [comment]
Thanks John.

**Pull request merged**